src/Files.App/Actions/Git/GitSyncAction.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Files.App.Commands;
+using Files.App.Contexts;
+
+namespace Files.App.Actions
+{
+	internal class GitSyncAction : ObservableObject, IAction
+	{
+		private readonly IContentPageContext _context;
+
+		public string Label { get; } = "GitSync".GetLocalizedResource();
+
+		public string Description { get; } = "GitSyncDescription".GetLocalizedResource();
+
+		public RichGlyph Glyph { get; } = new("\uEDAB");
+
+		public bool IsExecutable =>
+			_context.CanExecuteGitAction;
+
+		public GitSyncAction()
+		{
+			_context = Ioc.Default.GetRequiredService<IContentPageContext>();
+
+			_context.PropertyChanged += Context_PropertyChanged;
+		}
+
+		public Task ExecuteAsync()
+		{
+			return GitHelpers.PullOrigin(_context.ShellPage?.InstanceViewModel.GitRepositoryPath)
+				.ContinueWith(t => GitHelpers.PushToOrigin(
+					_context.ShellPage?.InstanceViewModel.GitRepositoryPath,
+					_context.ShellPage?.InstanceViewModel.GitBranchName));
```

Comment:
Could be made shorter by putting `_context.Shellpage?.InstanceViewModel` in a variable.